frontend/src/components/AcmDataForm.tsx

```diff
@@ -134,6 +138,29 @@ export function AcmDataFormPage(props: AcmDataFormProps): JSX.Element {
     <span style={{ wordBreak: 'keep-all' }}>{t('Copy to clipboard')}</span>
   )
 
+  // prevent navigating away from dirty form
+  // also see <Prompt> below
+  const resources = formData.stateToData()
+  if (!dataRef.current) {
+    dataRef.current = resources
+  } else {
+    dirtyRef.current = !isEqual(dataRef.current, resources)
+  }
+  useEffect(() => {
+    const preventUnload = (event: BeforeUnloadEvent) => {
+      if (dirtyRef.current) {
+        // NOTE: This message isn't used in modern browsers, but is required
+        const message = 'Sure you want to leave?'
```

Comment:
Might as well use `t('changes.maybe.lost')` here just in case?

Reply:
fixed the nit